### PR TITLE
chore: check code format on ci

### PR DESCRIPTION
**Pull request by araujogui**
## Description

Check code formatting on CI

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npx turbo lint` to ensure the code follows the style guide. And run `npx turbo lint:fix` to fix the style errors if necessary.
- [x] I have run `npx turbo format` to ensure the code follows the style guide.
- [x] I have run `npx turbo test` to check if all tests are passing, and/or `npx turbo test:snapshot` to update snapshots if I created and/or updated React Components.
- [x] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

**Review by AugustinMauroy — APPROVED** [review]
LGTM

File: .github/workflows/pull-request.yml
Comment (ovflowd) [inline .github/workflows/pull-request.yml]: Why would we want to run prettier on CI? 🤔

Comment (araujogui) [reply .github/workflows/pull-request.yml]: To check if there is any code formatting issue.
`npx turbo prettier` only checks that files are already correctly formatted, it doesn't overwrite them.

Comment (ovflowd) [reply .github/workflows/pull-request.yml]: Then the name of the job should not be changed, as it is not formatting the files and still checking the formatting.

Comment (ovflowd) [reply .github/workflows/pull-request.yml]: Can we revert the name of the job, then? prettier is part of the linting checks

**Review by mikeesto — COMMENTED** [review]
Looks good. Somewhat ironically, the PR check is failing because of a couple of prettier errors in two unrelated files.

File: .github/workflows/pull-request.yml
Comment (HinataKah0) [inline .github/workflows/pull-request.yml]: Nit:
For being consistent, shall we make the first letter uppercase?
```suggestion
      - name: Run Prettier
```

**Review by HinataKah0 — COMMENTED** [review]
LGTM! Let's run it once before merging

**Comment by ovflowd** [comment]
Can we run prettier on this PR? 🙈

File: public/blog-posts-data.json
Comment (ovflowd) [inline public/blog-posts-data.json:1]: The precommit hook should probably be updated (with the part that we store the `blog-posts-data.json`, can you do that?

File: .github/workflows/pull-request.yml
Comment (ovflowd) [inline .github/workflows/pull-request.yml]: No need to have this line of NEXT_TELEMETRY

File: .github/workflows/pull-request.yml
Comment (ovflowd) [inline .github/workflows/pull-request.yml]: Neither here (No need to have this line of NEXT_TELEMETRY)

File: .github/workflows/pull-request.yml
Comment (ovflowd) [inline .github/workflows/pull-request.yml:1]: Can we remove all the `NEXT_TELEMETRY_DISABLED: 1` except the one on the build step?

**Review by ovflowd — APPROVED** [review]
LGTM!

**Comment by ovflowd** [comment]
@araujogui prettier is still failing xD

File: .github/workflows/pull-request.yml
Comment (Harkunwar) [inline .github/workflows/pull-request.yml:47]: @ovflowd @araujogui  I have figured out why this Job is failing. 

We're running the lint stage before, which causes the `npm run lint:js` to execute, which modified the empty `blog-post-data.json` with data in the form of minified json. So when we run the prettier stage after, that file isn't in the pretty JSON format, and therefore this CI fails.

Easy fix:
Reorder the above stages to run Prettier before Linting.

Alternate fix:
Update `/Users/harkunwar/coding/nodejs.org/next-data/generateBlogPostsData.mjs` to save a formatted JSON when writing to `blog-post-data.json` using `JSON.stringify(..., null, 2)`. (Edit: I tested this, still doesn't make prettier happy, it wants a little different formatting compared to default JS)

Comment (ovflowd) [reply .github/workflows/pull-request.yml:47]: But shouldn’t prettier and eslint not conflict with each other?

Comment (ovflowd) [reply .github/workflows/pull-request.yml:47]: ESLint should be disabled for any file on the public folder. ESlint should not be complaining on these files…

Comment (Harkunwar) [reply .github/workflows/pull-request.yml:47]: @ovflowd, They shouldn't, but in this case we are causing a conflict because we're creating a file at the lint check stage using `generateBlogPostsData.mjs`. I don't even know why we're doing that.

Comment (ovflowd) [reply .github/workflows/pull-request.yml:47]: Why is generateBlogData being… Ah, because of “next lint” command?

We might need to introduce a check on next.config.mjs…

Or we should make a “prebuild” script to actually run the stuff we use for file generation.

Do you follow me? Can you make a quick PR? Probably a “prebuild” package.json step is easier.

Then we remove the lines of generating file from next.config to a dedicate file for these file generation. Like next.data.mjs

Comment (Harkunwar) [reply .github/workflows/pull-request.yml:47]: Once we commit any thing, the pre commit hooks run, which reset `blog-post-data.json` to `{ "pagination": [], "categories": [], "posts": [] }`, so there's nothing wrong in the commit, and all the files are formatted correctly.

If we run prettier before, it would always check `{ "pagination": [], "categories": [], "posts": [] }` and pass.

Comment (Harkunwar) [reply .github/workflows/pull-request.yml:47]: After running Lint, the file looks more like this
image

Comment (ovflowd) [reply .github/workflows/pull-request.yml:47]: Read my last comment 👀

Comment (Harkunwar) [reply .github/workflows/pull-request.yml:47]: Another easy workaround would be to just ignore this file.

Comment (ovflowd) [reply .github/workflows/pull-request.yml:47]: I think we need to fix regardless of this the next.config issue 😅

Comment (Harkunwar) [reply .github/workflows/pull-request.yml:47]: @ovflowd, I'm sorry something went wrong with my GitHub and didn't show your comments until now. I agree with what you propose.

**Comment by Harkunwar** [comment]
Closed by mistake, sorry.

**Review by Harkunwar — APPROVED** [review]
LGTM!

**Pull request merged**
